### PR TITLE
Update errormap.jl

### PR DESCRIPTION
**Pull request by sbromberger**
Fix deprecation warning in errormap.

### CODE REVIEW

**Comment by sbromberger** [comment]
See #23 and #24 - this is a more specific Dict definition.

**Comment by sbromberger** [comment]
So, I've managed to fix the deprecation warnings, but I note that the tests fail on 0.4 since the stream appears to be closed. This is unrelated to this PR.

File: src/errormap.jl [hunk 1]
```diff
@@ -1,6 +1,6 @@
-	# Error maps (maps err to (NAME, Description))
+        # Error maps (maps err to (NAME, Description))
 
-const error_codes = {
+const error_codes = Dict (
```
Comment (milktrader) [inline src/errormap.jl]: So is the idea that you use the most current syntax and Compat takes care of the rest, magically in the background?

**Comment by sbromberger** [comment]
The failed tests have nothing to do with the code changes here (though the tests probably need some rewriting since they depend on availability of external resources).

File: src/errormap.jl (hunk 1 above)
Comment (sbromberger) [inline src/errormap.jl]: That's basically it. For Dicts, I think you need `@compat` there but I can't tell you why.

Comment (milktrader) [inline src/errormap.jl]: Pretty sure it explicitly defines the two `Dict` syntaxes.

Comment (sbromberger) [inline src/errormap.jl]: It fails without @compat in 0.3:

```
julia> using Compat

julia> a = Dict("a"=>1)
ERROR: unsupported or misplaced expression =>

julia> a = @compat Dict("a"=>1)
Dict{ASCIIString,Int64} with 1 entry:
  "a" => 1
```

**Pull request merged**